app/models/user.rb

```diff
@@ -18,6 +18,7 @@ class User < ApplicationRecord
   validates_uniqueness_of :email, case_sensitive: false, allow_blank: false
 
   validates_exclusion_of :username, in: Blacklist.words
+  validates_format_of :username, with: /\A[A-Za-z0-9_]+\z/
```

Comment:
lets do this one, so you can't pick with something URL invalid like all numbers just in case?

```USERNAME_RE = /\A([a-z])([a-z0-9\-_]*)\z/i.freeze```

Reply:
@jake is screenhole.net/69 that invalid as a URL though? this is for users, not hole subdomains

Reply:
hm yea i guess your regex is fine, can always tighten later

Reply:
you sure? i'm easy either way